### ISSUE TITLE
Progress bar on `drush queue:run`

### ISSUE THREAD

**Issue by jigarius** [drush-ops/drush#5640]
**Is your feature request related to a problem? Please describe.**
When running a queue with `drush queue:run` it is not possible to see the overall progress of the queue.

**Describe the solution you'd like**
Ideally, we'd see a progress bar just like we see during `drush migrate:import`.

**Describe alternatives you've considered**
I usually have to run something like this in another terminal to track the progress:

```shell
watch -n 5 drush queue:list
```

**Comment by Ambient-Impact** [comment]
Agreed, this would be nice to have. It's also worth mentioning that if you run `drush queue:run <queue name> --verbose` you'll get a message printed for every queue item when it's completed like so:

```
 [info] Processing item 43875557 from warmer queue.
 [info] Processing item 43875558 from warmer queue.
 [info] Processing item 43875559 from warmer queue.
 [info] Processing item 43875560 from warmer queue.
 [info] Processing item 43875561 from warmer queue.
```

But it doesn't show overall progress so it's only so useful.

**Comment by weitzman** [comment]
if a queue has n items in it, are all calculations done with that number, even if --time-limit or --items-limit are provided?

**Comment by Ambient-Impact** [comment]
@weitzman Good question. This is one of those tricky UX issues that doesn't always have a clear answer. My suggestion:

- If no `--time-limit` or `--items-limit` are specified, show the total in the queue as a whole, but label it clearly as such.
- If `--items-limit` is specified, show both the current out of the items limit and of the total in the queue, like "Processed 3 out 10 in current limit; 58 remaining in queue".
- If `--time-limit` is specified, show the remaining time after each item is processed, and also show the remaining number in the queue like I suggest for `--items-limit`.
- Not sure what to do if both `--time-limit` and `--items-limit` are specified; does Drush allow this?
